osf/migrations/0037_auto_20170227_1522.py

```diff
@@ -0,0 +1,24 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.9 on 2017-02-27 21:22
+from __future__ import unicode_literals
+
+from django.db import migrations, connection
+
+
+def add_partial_uniqueness_index(*args):
+    sql = """
+    CREATE UNIQUE INDEX one_bookmark_collection_per_user ON osf_abstractnode (creator_id, is_bookmark_collection, is_deleted)
+        WHERE is_bookmark_collection=TRUE AND is_deleted=FALSE;
+    """
+    with connection.cursor() as cursor:
+        cursor.execute(sql)
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('osf', '0036_merge'),
+    ]
+
+    operations = [
+        migrations.RunPython(add_partial_uniqueness_index)
```

Comment:
The second parameter will allow you to apply a reverse migration to remove the index. Always a good idea to write reverse migrations.